crates/iota-proxy/src/peers.rs

```diff
@@ -128,184 +111,75 @@ impl IotaNodeProvider {
         None
     }
 
-    /// Get a mutable reference to the allowed iota validator map
+    /// Get a mutable reference to the allowed validator map
     pub fn get_mut(&mut self) -> &mut IotaPeers {
-        &mut self.nodes
+        &mut self.active_validator_nodes
     }
-
-    /// get_validators will retrieve known validators
-    async fn get_validators(url: String) -> Result<IotaSystemStateSummary> {
-        let rpc_method = "iotax_getLatestIotaSystemState";
-        let observe = || {
-            let timer = JSON_RPC_DURATION
-                .with_label_values(&[rpc_method])
-                .start_timer();
-            || {
-                timer.observe_duration();
-            }
-        }();
-        let client = reqwest::Client::builder().build().unwrap();
-        let request = serde_json::json!({
-            "jsonrpc": "2.0",
-            "method":rpc_method,
-            "id":1,
-        });
-        let response = client
-            .post(url)
-            .header(reqwest::header::CONTENT_TYPE, "application/json")
-            .body(request.to_string())
-            .send()
-            .await
-            .with_context(|| {
-                JSON_RPC_STATE
-                    .with_label_values(&[rpc_method, "failed_get"])
-                    .inc();
-                observe();
-                "unable to perform json rpc"
-            })?;
-
-        let raw = response.bytes().await.with_context(|| {
-            JSON_RPC_STATE
-                .with_label_values(&[rpc_method, "failed_body_extract"])
-                .inc();
-            observe();
-            "unable to extract body bytes from json rpc"
-        })?;
-
-        #[derive(Debug, Deserialize)]
-        struct ResponseBody {
-            result: IotaSystemStateSummary,
-        }
-
-        let body: ResponseBody = match serde_json::from_slice(&raw) {
-            Ok(b) => b,
-            Err(error) => {
-                JSON_RPC_STATE
-                    .with_label_values(&[rpc_method, "failed_json_decode"])
-                    .inc();
-                observe();
-                bail!(
-                    "unable to decode json: {error} response from json rpc: {:?}",
-                    raw
-                )
-            }
-        };
-        JSON_RPC_STATE
-            .with_label_values(&[rpc_method, "success"])
-            .inc();
-        observe();
-        Ok(body.result)
+    fn update_active_validator_set(&self, summary: &IotaSystemStateSummary) {
+        let validators = extract_validators_from_summaries(&summary.active_validators);
+        let mut allow = self.active_validator_nodes.write().unwrap();
+        allow.clear();
+        allow.extend(validators);
```

Comment:
does it make sense to just set `allow` to `validators` ? instead of clear() then extend() ?

```suggestion
    *allow = validators.collect();
```

same for pending validators below

Reply:
I would keep as is, `clear` will conserve the capacity of the vector, allowing `extend` to not allocate, unlike `collect`